File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
answer = 0
def DFS(nums, idx, sum, target):
    global answer
    if idx == len(nums) and sum == target:
        answer += 1
        return
    if idx == len(nums):
        return
    DFS(nums, idx+1, sum+nums[idx], target)
    DFS(nums, idx+1, sum-nums[idx], target)

def solution(numbers, target):
    global answer
    DFS(numbers, 0, 0, target)
    return answer

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.swift
import Foundation

func solution(_ numbers:[Int], _ target:Int) -> Int {
    var result = 0
    func DFS(idx: Int, num: Int) {
        if idx == numbers.count {
            if num == target {
                result += 1
            }
            return
        }
        DFS(idx: idx+1, num: num + numbers[idx])
        DFS(idx: idx+1, num: num - numbers[idx])
    }
    DFS(idx: 0, num: 0)
    return result
}